Deprecate injected bundle frame access interfaces

achristensen07 opened the pull request:
#### 5fa09b4d3b192cb25cff5aad2f8fe41c3210dad5
<pre>
Deprecate injected bundle frame access interfaces
<a href="https://bugs.webkit.org/show_bug.cgi?id=276554">https://bugs.webkit.org/show_bug.cgi?id=276554</a>
<a href="https://rdar.apple.com/131635665">rdar://131635665</a>

Reviewed by Carlos Garcia Campos.

With site isolation, even the main frame might not be in the current process if the current process
is the process hosting cross-site iframe content.  WKBundleFrameGetParentFrame and WKBundleFrameCopyChildFrames
are already deprecated, but this needs to be done too.  It would be nice to deprecate
webkit_web_page_get_main_frame as well, but I don&apos;t maintain that API.  I&apos;ll bring it up in the mailing list.

* Source/WebKit/WebProcess/InjectedBundle/API/c/WKBundlePage.h:
* Source/WebKit/WebProcess/InjectedBundle/API/mac/WKWebProcessPlugInBrowserContextController.h:

Canonical link: <a href="https://commits.webkit.org/281688@main">https://commits.webkit.org/281688@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f838adbca034ed84802f2bd0eaf615615a93a8d9

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/60531 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/39888 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/13101 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/64458 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/11074 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/62661 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/47564 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/11300 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/48969 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/7700 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/62564 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/37155 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/52432 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/29801 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/33846 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/9665 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/9987 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/55718 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/9965 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/66188 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/4470 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/9775 "layout-tests (failure)") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/56336 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/4491 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/52413 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/56507 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/3708 "Passed tests") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/9122 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/35694 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/36776 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/37867 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/36521 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

carlosgcampos commented on Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitWebPage.h.in:
```diff
@@ -93,7 +93,7 @@ webkit_web_page_get_id                      (WebKitWebPage *web_page);
 webkit_web_page_get_uri                     (WebKitWebPage *web_page);
 
-WEBKIT_API WebKitFrame *
+WEBKIT_DEPRECATED WebKitFrame *
```
This is not enough. You also need to add Deprecated tag to the api docs, in `WebKitWebPage.cpp` add:

```
* Deprecated: 2.46
```

After the Since tag.

You also need to add ignore deprecation guards (G_GNUC_BEGIN_IGNORE_DEPRECATIONS/G_GNUC_END_IGNORE_DEPRECATIONS) in the api tests using the function. I think EWS isn't failing because this is only used in tests running already deprecated APIs that are only built for GTK3 and EWS builds with GTK4.

achristensen07 committed:
Deprecate injected bundle frame access interfaces
https://bugs.webkit.org/show_bug.cgi?id=276554
rdar://131635665

Reviewed by Carlos Garcia Campos.

With site isolation, even the main frame might not be in the current process if the current process
is the process hosting cross-site iframe content.  WKBundleFrameGetParentFrame and WKBundleFrameCopyChildFrames
are already deprecated, but this needs to be done too.  It would be nice to deprecate
webkit_web_page_get_main_frame as well, but I don't maintain that API.  I'll bring it up in the mailing list.

* Source/WebKit/WebProcess/InjectedBundle/API/c/WKBundlePage.h:
* Source/WebKit/WebProcess/InjectedBundle/API/mac/WKWebProcessPlugInBrowserContextController.h:

Canonical link: https://commits.webkit.org/281688@main
